packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+    public name?: string;
+    /**
+     * A preview of the file which is a lightweight representation.
+     * In case of images this will be a thumbnail/compressed image in base64 encoding.
+     */
+    public preview: string;
+  }
+
+  /**
+   * Checks whether or not visualMedia has user permission
+   *
+   * @beta
+   * @returns Promise that will resolve with true if the user had granted the app permission to media information, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.has', permissions));
+    });
+  }
+
+  /**
+   * Requests user permission for visualMedia
+   * @beta
+   * @returns Promise that will resolve with true if the user consented permission for media, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.request', permissions));
+    });
+  }
+
+  /**
+   * Checks if visualMedia capability is supported by the host
+   * @returns boolean to represent whether media is supported
+   * @beta
+   * @throws Error if {@linkcode app.initialize} has not successfully completed
+   */
+  export function isSupported(): boolean {
+    return ensureInitialized(runtime) && runtime.supports.visualMedia && runtime.supports.permissions ? true : false;
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    /**
+     * Capture one or multiple image(s) throughing camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     */
+    export function captureImages(imageInputs: ImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!imageInputs || imageInputs == null || imageInputs.visualMediaCount > 10) {
+        throw new Error('Must supply the valid image(s)');
```

Comment:
>'Must supply the valid image(s)' [](http://example.com/codeflow?start=24&length=32)

This error message does not seem to align with the condition that is being tested or the semantics of this function. As far as I can tell, developers are not expected to supply valid images to this function (they receive valid images back), so telling them that they must supply valid images doesn't seem clear.

Reply:
Updated the error message to specify mentioned the maxVisualMediaCount.